### ISSUE TITLE
[Feature]: change the draw_debug_frame position and rotation

### ISSUE THREAD

**Issue by ucacaxm** [Genesis-Embodied-AI/Genesis#1853]
### What feature or enhancement are you proposing?

Hi,

I would like to change the position and orientation of a `draw_debug_frame` during simulation.

I haven't found any examples, and the type `genesis.ext.pyrender.mesh.Mesh` suggests that it is not multiplied by a matrix?

It could be interesting to be able to do that for any debug object.

Thanks,

### Motivation

I need to visualize how the coordinate/orientation change during time.

### Potential Benefit

This would significantly increase debugging capabilities.

### What is the expected outcome of the implementation work?

- [ ]

**Comment by duburcqa** [comment]
See unit test `tests/test_render.py::test_draw_debug`. Here is a snippet:
```python
poses = gu.trans_quat_to_T(2.0 * (np.random.rand(3, 3) - 0.5), np.random.rand(3, 4))
scene.visualizer.context.update_debug_objects([frame_obj, sphere_obj], poses)
```

**Comment by duburcqa** [comment]
Let me know if it does not work and I will reopen.

**Comment by ucacaxm** [comment]
Thanks.

I tested the code  that I added at the end of the post  and got a code dump:

```
Thread 1 "python" received signal SIGSEGV, Segmentation fault.
0x00007ffe752a31a9 in glBindBuffer () from /lib/x86_64-linux-gnu/libOpenGL.so.0
(gdb) bt
#0  0x00007ffe752a31a9 in glBindBuffer () from /lib/x86_64-linux-gnu/libOpenGL.so.0
#1  0x00007ffe7a727054 in ?? ()
#2  0x00007fffffffdac8 in ?? ()
#3  0x0000000000000000 in ?? ()
(gdb) 
```
Usually no problem with my computer and OpenGL/driver/


If I comment `scene.visualizer.context.update_debug_objects([frame_obj, sphere_obj], poses)`
it is ok (but the frame does not move).


My setup
```
[Genesis] [08:36:37] [INFO] Running on [NVIDIA GeForce RTX 3060] with backend gs.cuda. Device memory: 11.75 GB.
[Genesis] [08:36:37] [INFO] 🚀 Genesis initialized. 🔖 version: 0.3.4, 🌱 seed: 0, 📏 precision: '32', 🐛 debug: False, 🎨 theme: 'dark'.
```

The tested code:

```
import time
import genesis as gs
import numpy as np



if __name__ == "__main__":
    ########################## init ##########################
    #    gs.init(seed=0, backend=gs.cpu if args.cpu else gs.gpu)
    gs.init(seed=0, backend=gs.gpu) 

    ########################## create a scene ##########################
    scene = gs.Scene(
            sim_options=gs.options.SimOptions(
            substeps=2,
            gravity=(0, 0, -9.8),
        ),
        viewer_options=gs.options.ViewerOptions(
            camera_pos=(-5.0, -5.0, 2.0),
            camera_lookat=(0.0, 0.0, 0.0),
            camera_fov=40,
        ),
        show_viewer=True,
        rigid_options=gs.options.RigidOptions(
            dt=0.01,
            constraint_solver=gs.constraint_solver.Newton,
        ),
    )

    scene.add_entity(
        morph=gs.morphs.Plane(),
        material=gs.materials.Rigid(
            coup_friction=5.0,
        ),
    )

    scene.build(n_envs=1)


    scene.draw_debug_arrow(
        pos=(0, 0.4, 0.1),
        vec=(0, 0.3, 0.8),
        color=(1, 0, 0),
    )
    scene.draw_debug_line(
        start=(0.7, -0.3, 0.7),
        end=(0.6, 0.2, 0.7),
        radius=0.01,
        color=(1, 0, 0, 1),
    )
    sphere_obj = scene.draw_debug_sphere(
        pos=(-0.3, 0.3, 0.0),
        radius=0.15,
        color=(0, 1, 0),
    )
    frame_obj = scene.draw_debug_frame(
        T=np.array(
            [
                [1.0, 0.0, 0.0, -0.3],
                [0.0, 0.0, 1.0, 0.0],
                [0.0, -1.0, 0.0, 1.2],
                [0.0, 0.0, 0.0, 1.0],
            ]
        ),
        axis_length=1.0,
        origin_size=0.03,
        axis_radius=0.05,
    )



    stop = False 

    while not stop:
        poses = gs.trans_quat_to_T(2.0 * (np.random.rand(3, 3) - 0.5), np.random.rand(3, 4))    
        scene.visualizer.context.update_debug_objects([frame_obj, sphere_obj], poses)

        scene.step()
```

**Comment by ucacaxm** [comment]
I tested it on another computer and got the same error.

[Genesis] [11:59:27] [INFO] Consider setting 'performance_mode=True' in production to maximise runtime speed, if significantly increasing compilation time is not a concern.
[Genesis] [11:59:27] [INFO] Running on [NVIDIA GeForce RTX 4090] with backend gs.cuda. Device memory: 23.64 GB.
[Genesis] [11:59:27] [INFO] 🚀 Genesis initialized. 🔖 version: 0.3.4, 🌱 seed: 0, 📏 precision: '32', 🐛 debug: False, 🎨 theme: 'dark'.

**Comment by duburcqa** [comment]
>  poses = gs.trans_quat_to_T(2.0 * (np.random.rand(3, 3) - 0.5), np.random.rand(3, 4))    

This is for a batch size of 3. I should probably add a check and raise a proper exception.

EDIT: Hum... I double checked and it should be fine. I will try that later.

**Comment by ucacaxm** [comment]
I saw the problem of 2 objects and 3 matrices. I tested

```
   while not stop:
        poses = gs.trans_quat_to_T(2.0 * (np.random.rand(3, 3) - 0.5), np.random.rand(3, 4))    
        scene.visualizer.context.update_debug_objects([frame_obj, sphere_obj, arrow_obj], poses)
```
but same core dump


I also supposed that the problem arise because of the form of the random matrices (non invertible matrix).
I tested this 

```
    stop = False 
    time = 0.0
    while not stop:
        axis = torch.tensor([0.0, 0.0, 1.0]) 
        angle = torch.cos( torch.tensor(time) )
        quat = gs.axis_angle_to_quat(angle, axis)
        pos = torch.tensor([3.0, 3.0, 3.0]) 
        pose = gs.trans_quat_to_T( trans=pos, quat = quat)
        print(pose)
        scene.visualizer.context.update_debug_objects([frame_obj], [pose])
        scene.step()
        time += scene.dt
```
but same core dump ...

**Comment by ucacaxm** [comment]
I checked the code in `rasterizer_context.py`

The function `get_buffer_debug_objects`returns a well formatted dict with `19` and my matrix.
19 seems coherent as GL buffer Id.

I checked the code of the 2  functions `update_buffer`  and it seems ok.

But with GDB, when arise the core dump, `glBindBuffer` get 11 as buffer id:
```
(gdb) bt
#0  0x00007ffe391c31a9 in glBindBuffer () from /lib/x86_64-linux-gnu/libOpenGL.so.0
#1  0x00007ffea0e6e054 in ?? ()
#2  0x00007fffffffdab8 in ?? ()
#3  0x0000000000000000 in ?? ()
(gdb) info registers rdi rsi
rdi            0x8892              34962
rsi            0xb                 11
```
rdi = 0x8892 = GL_ARRAY_BUFFER
rsi = 11 


I have to find a way to debug `update_buffer` in `jit_render.py` ...

**Comment by duburcqa** [comment]
I think I know why it is crashing. Could you try with:
```python
        viewer_options=gs.options.ViewerOptions(
            run_in_thread=False,
        ),
```

**Comment by ucacaxm** [comment]
With True, core dump
With False, it is ok => except that the provided matrix is multiplied by the initial one which is a bit disturbing


Before that I installed Genesis with `pip -e`to debug.
I focus on this line:
`buffer_updates[self._scene.get_buffer_id(node, "model")] = pose.transpose((0, 2, 1))`
from `rasterized_context.py`line 919

This part `self._scene.get_buffer_id(node, "model")`returns sometime -1 and sometime another wrong number.
=> since there are 2 threads, shared data goes wrong ...

**Comment by duburcqa** [comment]
I'm currently investigating this issue. The current implementation is buggy and very fragile. I will fix what can be fixed, and forbid everything else.

**Comment by ucacaxm** [comment]
I confirm this code bellow is ok.
But with `run_in_thread=True` => core dump.

I wanted to help and find a fix, but with the thread aspect it is probably to deeply embedded in the viewer.

Thanks for your help


```
import time
import genesis as gs
import numpy as np
import torch


if __name__ == "__main__":

    gs.init(seed=0, backend=gs.gpu) 

    scene = gs.Scene(
            sim_options=gs.options.SimOptions(
            substeps=2,
            gravity=(0, 0, -9.8),
        ),
        viewer_options=gs.options.ViewerOptions(
            camera_pos=(-5.0, -5.0, 2.0),
            camera_lookat=(0.0, 0.0, 0.0),
            camera_fov=40,
            run_in_thread=False,
        ),
        show_viewer=True,
        rigid_options=gs.options.RigidOptions(
            dt=0.01,
            constraint_solver=gs.constraint_solver.Newton,
        ),
    )


    scene.add_entity(
        morph=gs.morphs.Plane(),
        material=gs.materials.Rigid(
            coup_friction=5.0,
        ),
    )

    scene.build(n_envs=1)


    frame_obj = scene.draw_debug_frame(
        T=np.array(
            [
                [1.0, 0.0, 0.0,  0.0],
                [0.0, 1.0, 0.0,  0.0],
                [0.0, 0.0, 1.0,  0.0],
                [0.0, 0.0, 0.0,  1.0],
            ]
        ),
        axis_length=1.0,
        origin_size=0.03,
        axis_radius=0.05,
    )

    stop = False 
    time = 0.0
    while not stop:
        axis = torch.tensor([0.0, 0.0, 1.0])  
        angle = 3.14 * torch.cos( torch.tensor(time) )
        quat = gs.axis_angle_to_quat(angle, axis) 
        pos = torch.tensor([1.0, 1.0, 1.0])  
        pose = gs.trans_quat_to_T( trans=pos, quat = quat)
        scene.visualizer.context.update_debug_objects([frame_obj], [pose])
        scene.step()
        time += scene.dt
    ```

**Comment by duburcqa** [comment]
Here is the "fix": https://github.com/Genesis-Embodied-AI/Genesis/pull/1869/commits/776e3ca3af40f65fbf93daf769641f45d0bcf968

I had to disable support of many primitive shapes. Feel free to open a PR that adds the support of the missing one if you want.
